server/app/services/grid_service_instance_deployer.rb

```diff
@@ -32,7 +33,7 @@ def deploy(node, instance_number, deploy_rev, creds = nil)
   # @param [String] deploy_rev
   def wait_for_service_to_start(node, instance_number, deploy_rev)
     # node/agent has 5 minutes to do it's job
-    Timeout.timeout(300) do
+    wait!(timeout: 300) do
       sleep 0.5 until self.deployed_service_container_exists?(instance_number, deploy_rev)
```

Comment:
This looks broken... it's `sleep until ...` within the wait-block, and does this actually return `true`?

This does not ever time out anymore if the service fails to deploy... which is a problem, because the 1.1 RPC's don't report errors either.